### ISSUE TITLE
RDAdmin: Manage Feed Crashes

### ISSUE THREAD

**Issue by alexolivan** [ElvishArtisan/rivendell#820]
Hi.
I've been messing with Podcast Feeds for my recordings... and from time to time, RDAdmin is crashing, both adding and editing. Is not consistent, but from time to time.
Changes at DB seem to be consistently persisted. I'm inclined to think on some frontend/Qt5 issue:

Here's DBG backtrace

``` 
ASSERT failure in QList<T>::operator[]: "index out of range", file /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h, line 579
--Type <RET> for more, q to quit, c to continue without paging--c

Thread 1 "rdadmin" received signal SIGABRT, Aborted.
__GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
50	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) bt
#0  __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:50
#1  0x00007ffff53f0537 in __GI_abort () at abort.c:79
#2  0x00007ffff5aa0bf1 in QMessageLogger::fatal(char const*, ...) const ()
   from /lib/x86_64-linux-gnu/libQt5Core.so.5
#3  0x00007ffff5aa009c in qt_assert_x(char const*, char const*, char const*, int) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#4  0x00007ffff77a03f4 in QList<QVariant>::operator[] (i=<optimized out>, 
    this=<optimized out>)
    at /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:579
#5  0x00007ffff77a086b in QList<QList<QList<QVariant> > >::operator[] (
    i=<optimized out>, this=<optimized out>)
    at /usr/include/x86_64-linux-gnu/qt5/QtCore/qlist.h:364
#6  RDFeedListModel::updateRow (this=0x555555adea10, row=5, q=0x555555be3f20)
    at rdfeedlistmodel.cpp:582
#7  0x00007ffff78be4fe in RDFeedListModel::updateRowLine (this=0x555555adea10, 
    line=5) at rdfeedlistmodel.cpp:538
#8  0x00007ffff78be674 in RDFeedListModel::refreshRow (this=0x555555adea10, 
    index=...)
    at /usr/include/x86_64-linux-gnu/qt5/QtCore/qabstractitemmodel.h:62
#9  0x0000555555632ce7 in ListFeeds::editData() ()
#10 0x00007ffff5cf35e0 in ?? () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#11 0x00007ffff6a32e42 in QAbstractButton::clicked(bool) ()
   from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
--Type <RET> for more, q to quit, c to continue without paging--c
#12 0x00007ffff6a337aa in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#13 0x00007ffff6a350ff in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#14 0x00007ffff6a352d1 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#15 0x00007ffff6985fae in QWidget::event(QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#16 0x00007ffff694615f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#17 0x00007ffff694d04b in QApplication::notify(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#18 0x00007ffff5cbcfca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#19 0x00007ffff694c076 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#20 0x00007ffff699e011 in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#21 0x00007ffff69a133e in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#22 0x00007ffff694615f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#23 0x00007ffff5cbcfca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#24 0x00007ffff625e583 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#25 0x00007ffff623464c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#26 0x00007ffff15c97da in ?? () from /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#27 0x00007ffff43d7e6b in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#28 0x00007ffff43d8118 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#29 0x00007ffff43d81cf in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#30 0x00007ffff5d1451f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#31 0x00007ffff5cbb98b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#32 0x00007ffff6b3ee77 in QDialog::exec() () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#33 0x0000555555649e1a in MainWidget::podcastsData() ()
#34 0x00007ffff5cf35e0 in ?? () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#35 0x00007ffff6a32e42 in QAbstractButton::clicked(bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#36 0x00007ffff6a337aa in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#37 0x00007ffff6a350ff in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#38 0x00007ffff6a352d1 in QAbstractButton::mouseReleaseEvent(QMouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#39 0x00007ffff6985fae in QWidget::event(QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#40 0x00007ffff694615f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#41 0x00007ffff694d04b in QApplication::notify(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#42 0x00007ffff5cbcfca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#43 0x00007ffff694c076 in QApplicationPrivate::sendMouseEvent(QWidget*, QMouseEvent*, QWidget*, QWidget*, QWidget**, QPointer<QWidget>&, bool, bool) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#44 0x00007ffff699e011 in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#45 0x00007ffff69a133e in ?? () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#46 0x00007ffff694615f in QApplicationPrivate::notify_helper(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Widgets.so.5
#47 0x00007ffff5cbcfca in QCoreApplication::notifyInternal2(QObject*, QEvent*) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#48 0x00007ffff625e583 in QGuiApplicationPrivate::processMouseEvent(QWindowSystemInterfacePrivate::MouseEvent*) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#49 0x00007ffff623464c in QWindowSystemInterface::sendWindowSystemEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Gui.so.5
#50 0x00007ffff15c97da in ?? () from /lib/x86_64-linux-gnu/libQt5XcbQpa.so.5
#51 0x00007ffff43d7e6b in g_main_context_dispatch () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#52 0x00007ffff43d8118 in ?? () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#53 0x00007ffff43d81cf in g_main_context_iteration () from /lib/x86_64-linux-gnu/libglib-2.0.so.0
#54 0x00007ffff5d1451f in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#55 0x00007ffff5cbb98b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#56 0x00007ffff5cc3c00 in QCoreApplication::exec() () from /lib/x86_64-linux-gnu/libQt5Core.so.5
#57 0x00005555555b223d in main ()
```

Regards.

EDIT:
I think the problem happens while messing at the bottom of the list, since I've a bunch of feeds (I'm trying to learn how to use the superfeed stuff) and it seems the problem happens when handling the last feed at the bottom (So when there's just one or two feeds you trip the wire more often)

**Comment by ElvishArtisan** [comment]
Confirmed here.

**Comment by ElvishArtisan** [comment]
Fixed in 30e6494. Please test!
